src/main.rs

```diff
@@ -1,33 +1,40 @@
 // this program was a mistake. what am i doing
 use rand::Rng;
 use std::env;
-use std::process;
+use std::process::ExitCode;
 
-fn main() {
-    let meows = vec!["meow", "mrrrow", "mrrrp", "nyaaa", "nya", "mew", "meowwww"];
-    let args: Vec<_> = env::args().collect();
-    if args.len() <= 1 {
-        println!("meower help:\n\nusage: meower <amount>\nexample: meower 10");
-        process::exit(0x1);
-    }
-    let amount = args[1].parse::<i32>().unwrap()-1;
-    let mut i: i32 = 0;
-    let mut prevnum = 9999;
-    while i <= amount {
+fn main() -> ExitCode {
+    let meows = ["meow", "mrrrow", "mrrrp", "nyaaa", "nya", "mew", "meowwww"];
+    let amount = if let Some(amount) = env::args().skip(1).next() {
+        amount.parse::<usize>().unwrap()
+    } else {
+        println!(
+            "meower help:\n\
+            \n\
+            usage: meower <amount>\n\
+            example: meower 10"
+        );
+        return ExitCode::FAILURE;
+    };
+    let mut prevnum: Option<usize> = None;
+    for _ in 0..amount {
         let mut num = rand::thread_rng().gen_range(0..meows.len());
-        if num == prevnum {
-            if num > 3 {
-                num = num-1
-            } else if num < 3 {
-                num = num+1
+        if let Some(prevnum) = prevnum {
+            if num == prevnum {
+                num = if num > (meows.len() / 2) {
```

Comment:
i think i will do this (not gonna merge cuz i merged another one and they would conflict and it would just be all broken n things :( but ill take a look at some of these manually)